arrow-libs/core/arrow-continuations/src/commonTest/kotlin/generic/TestSuite.kt

```diff
@@ -4,97 +4,21 @@ import arrow.continuations.Reset
 import arrow.continuations.generic.RestrictedScope
 import arrow.core.Either
 import arrow.core.Either.Left
-import arrow.core.test.UnitSpec
+import io.kotest.core.spec.style.StringSpec
 import io.kotest.matchers.shouldBe
 
-abstract class ContTestSuite : UnitSpec() {
-  abstract suspend fun <A> runScope(func: (suspend RestrictedScope<A>.() -> A)): A
-
-  abstract fun capabilities(): Set<ScopeCapabilities>
-
-  init {
+class SingleShotContTestSuite : StringSpec({
     "yield a list (also verifies stacksafety)" {
-      runScope<List<Int>> {
+      Reset.restricted {
         suspend fun <A> RestrictedScope<List<A>>.yield(a: A): Unit = shift { k -> listOf(a) + k(Unit) }
         for (i in 0..10_000) yield(i)
         emptyList()
       } shouldBe (0..10_000).toList()
     }
     "short circuit" {
-      runScope<Either<String, Int>> {
+      Reset.restricted<Either<String, Int>> {
         val no: Int = shift { Left("No thank you") }
         throw IllegalStateException("This should not be executed")
       } shouldBe Left("No thank you")
     }
-
-    // some examples from http://homes.sice.indiana.edu/ccshan/recur/recur.pdf
-    if (capabilities().contains(ScopeCapabilities.MultiShot)) {
```

Comment:
The compile error in `arrow-continuations` was coming from here @serras.

Not entirely sure why, but perhaps a regression bug in the compiler due to `abstract class ContTestSuite` being combined with `abstract class UnitSpec` in combination with the `abstract fun capabilities()`. Might be interesting to share with Kotlin team(?).

As you can see `SingleShotContTestSuite` was the only implementer of `ContTestSuite`, and it was returning `emptySet` for `capabilities()`. So these tests were not being run, and they never ran. We left the code for reasons that don't matter anymore, so removing them is non-impactful and safe.

Reply:
I found this by opening the `js` file that was linked in the stack trace at runtime, `/var/folders/qs/9sb020z91_z1_50v18bnmf0000gn/T/_karma_webpack_466164/common.js:5773`. There I saw it was calling `ContTestSuite` and that led me to the `capabilities` function.

Reply:
I'm amazed you found it. Thanks very much! We're just missing the STM problem then...